LayoutTests/imported/w3c/web-platform-tests/compression/decompression-bad-chunks.tentative.any.serviceworker-expected.txt

```diff
@@ -1,4 +1,6 @@
 
+Harness Error (FAIL), message = Unhandled rejection: Failed to Decode Data.
```

Comment:
Which promise is actually not handled? readPromise maybe?
Can we change the test to not have this message?

Reply:
Going to take a look into that

Reply:
The most interesting thing I noticed is that this does not trigger in the other mutations (non service worker, shared worker, or worker). I wonder what makes service worker special.

Reply:
Still need to fix

Reply:
I think we should write the tests in this way:
1. writer.close() should be called and should reject so it is good to add the promise_rejects_js. Ditto for reader.closed.
2. writePromise may not always reject as we have seen, but we need to handle the promise in case it is rejecting. something like writePromise.then(() => { }, () => { });
3. reader.read() should probably reject. If it does not reject, it will mean we will have a bigger behavioural difference since some potentially bad data will be given to the application. If it always reject, we should add a promise_rejects_js assertion. Otherwise do then(() => {}, () => {}).

Reply:
I rewrote the tests to follow this pattern.

It would be worth having a discussion around the differences of when the error is thrown between the two frameworks, and what a developer's expectation would be in this case.